### PR TITLE
Make shifting Fields or local Detunings optional. Add tests.

### PR DESCRIPTION
**Pull request by ykharkov**
*Resolving Issue #22 *

*Description of changes:*
 - Support both `localDetuning` and `shiftingFields` as input fields in AHS program JSON.
 - Make these fields optional (ignore if not present in AHS JSON)
 - Add test for empty `localDetuning` or `shiftingFields` field

### CODE REVIEW

File: src/mps_utils.jl [hunk 1]
```diff
@@ -55,10 +55,26 @@ function parse_protocol(ahs_program, τ::Float64, n_τ_steps::Int)
...
     time_points_Ω = ahs_program["hamiltonian"]["drivingFields"][1]["amplitude"]["time_series"]["times"]
     values_Ω = ahs_program["hamiltonian"]["drivingFields"][1]["amplitude"]["time_series"]["values"]
-
-    pattern = ahs_program["hamiltonian"]["shiftingFields"][1]["magnitude"]["pattern"]
-    time_points_shift = ahs_program["hamiltonian"]["shiftingFields"][1]["magnitude"]["time_series"]["times"]
-    values_shift = ahs_program["hamiltonian"]["shiftingFields"][1]["magnitude"]["time_series"]["values"]
+        
+    n_atoms = length(ahs_program["setup"]["ahs_register"]["sites"])
+    # Supporting both localDetuning and shiftingFields naming conventions
+    loc_detuning_key = ""
```
Comment (kshyatt-aws) [inline src/mps_utils.jl:61]: Why not

```julia

loc_detuning_key = if "shiftingFields" ∈ keys(ahs_program["hamiltonian"])
        "shiftingFields"
    elseif "localDetuning" ∈ keys(ahs_program["hamiltonian"])
        "localDetuning"
    end
```

File: src/mps_utils.jl [hunk 2]
```diff
@@ -55,10 +55,26 @@ function parse_protocol(ahs_program, τ::Float64, n_τ_steps::Int)
...
+    if "shiftingFields" ∈ keys(ahs_program["hamiltonian"])
+        loc_detuning_key = "shiftingFields"
+    elseif "localDetuning" ∈ keys(ahs_program["hamiltonian"])
+        loc_detuning_key = "localDetuning"
+    end
+    
+    # Check if local detuning or shifting fields were found, otherwise assign zeros
+    if !isempty(loc_detuning_key)
```
Comment (kshyatt-aws) [inline src/mps_utils.jl:69]: I think you can simplify this with [`Base.get`](https://docs.julialang.org/en/v1/base/collections/#Base.get)

File: test/runtests.jl [hunk 3]
```diff
@@ -2,7 +2,7 @@ using BraketAHS: run, save_results
 using Test
 using JSON3
 
-@testset "BraketAHS.jl" begin
+@testset "BraketAHS.jl: Non-empty local detunings" begin
     # Write your tests here.
```
Comment (kshyatt-aws) [inline test/runtests.jl:6]: Maybe we can get rid of this comment lol

File: src/mps_utils.jl (hunk 1 above)
Comment (ykharkov) [reply src/mps_utils.jl]: Thanks, fixed

File: test/runtests.jl (hunk 3 above)
Comment (ykharkov) [reply test/runtests.jl]: Removed comment
